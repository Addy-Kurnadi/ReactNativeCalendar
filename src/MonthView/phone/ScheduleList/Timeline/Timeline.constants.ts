export const fullTimeArray = [
    "0000", "0100", "0200", "0300", "0400",
    "0500", "0600", "0700", "0800", "0900",
    "1000", "1100", "1200", "1300", "1400",
    "1500", "1600", "1700", "1800", "1900",
    "2000", "2100", "2200", "2300", "2400"
]

export const fullTimeArrayRender = [
    "00:00", "01:00", "02:00", "03:00", "04:00",
    "05:00", "06:00", "07:00", "08:00", "09:00",
    "10:00", "11:00", "12:00", "13:00", "14:00",
    "15:00", "16:00", "17:00", "18:00", "19:00",
    "20:00", "21:00", "22:00", "23:00", "24:00"
]
