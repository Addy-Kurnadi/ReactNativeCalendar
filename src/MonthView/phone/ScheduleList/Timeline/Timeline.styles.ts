import { StyleSheet, TextStyle, ViewStyle } from "react-native"

interface Style {
    outerViewContainer: ViewStyle
    textStyle: TextStyle
}

export default StyleSheet.create<Style>({
    outerViewContainer: {
        flex: 1,
    },
    textStyle: {
        textAlign: "center",
        fontSize: 14,
        fontWeight: "bold",
        color: "black",
    }
})
