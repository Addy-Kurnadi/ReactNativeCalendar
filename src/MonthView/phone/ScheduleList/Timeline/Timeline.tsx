import * as React from "react"
import { Text, View } from "react-native"
import { fullTimeArray, fullTimeArrayRender } from "./Timeline.constants"
import styles from "./Timeline.styles"

export type TimeType = "0000" | "0100" | "0200" | "0300" | "0400" | "0500" | "0600" | "0700" | "0800" | "0900" |
                "1000" | "1100" | "1200" | "1300" | "1400" | "1500" | "1600" | "1700" | "1800" | "1900" | "2000" |
                "2100" | "2200" | "2300" | "2400"

interface TimelineProps {
    startTime: TimeType
    endTime: TimeType
    height: number
}

export const Timeline = ({startTime, endTime, height}: TimelineProps) => {
    const indexOfStartTime = fullTimeArray.indexOf(startTime)
    const indexOfEndTime = fullTimeArray.indexOf(endTime)
    const keepTimeArray = fullTimeArrayRender.slice(indexOfStartTime, indexOfEndTime + 1)
    const halfOfHeight = height * 0.5
    const renderTextHandler = (time: string, index: number) => {
        return (
            <View key={index} style={{height}}>
                <Text style={styles.textStyle}>
                    {time}
                </Text>
            </View>
        )
    }

    return (
        <View style={styles.outerViewContainer}>
            <View style={{height: halfOfHeight}}/>
                {keepTimeArray.map(renderTextHandler)}
            <View style={{height: halfOfHeight}}/>
        </View>
    )
}
