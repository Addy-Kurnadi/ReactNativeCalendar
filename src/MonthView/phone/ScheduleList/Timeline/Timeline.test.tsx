import * as React from "react"
import { create } from "react-test-renderer"
import { Timeline } from "./Timeline"

describe("<Timeline />", () => {
    it("should render with given props", () => {
        const component = create(
            <Timeline
                startTime={"0800"}
                endTime={"1800"}
                height={50}
            />).toJSON()
        expect(component).toMatchSnapshot()
    })
})
