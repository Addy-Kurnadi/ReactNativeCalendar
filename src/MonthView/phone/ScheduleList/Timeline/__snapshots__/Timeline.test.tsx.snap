// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline /> should render with given props 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 25,
      }
    }
  />
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      08:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      09:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      10:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      11:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      12:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      13:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      14:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      15:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      16:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      17:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginTop": -7.5,
          "textAlign": "center",
        }
      }
    >
      18:00
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 25,
      }
    }
  />
</View>
`;
